cookbook/upgrade/major_version.rst

```diff
@@ -0,0 +1,112 @@
+.. index::
+    single: Upgrading; Major Version
+
+Upgrading a Major Version (e.g. 2.7.0 to 3.0.0)
+===============================================
+
+Once in a couple years, Symfony releases a new major version release (the
+first number changes). These releases are the trickiest to upgrade, as they are
+allowed to contain BC breaks. However, Symfony tries to make this upgrade
+process as smooth as possible.
+
+This means that you can update most of your code before the major release is
+actually released. This is called making your code *future compatible*.
+
+There are a couple of steps to upgrading a major version:
+
+#. :ref:`Make your code deprecation free <upgrade-major-symfony-deprecations>`;
+#. :ref:`Update to the new major version via Composer <upgrade-major-symfony-composer>`.
+#. :ref:`Update your code to work with the new version <upgrade-major-symfony-after>`
+
+.. _upgrade-major-symfony-deprecations:
+
+1) Make your Code Deprecation Free
+----------------------------------
+
+During a lifecycle of a major release, new features are added and method
+signatures and public API usages are changed. However, minor versions should
+not contain any backwards compatibility changes. It is made sure that there is
+a so-called *backwards compatibility layer* (or BC layer). This means that the
+old API will still work, while the new feature is used internally. This BC
+layer is then marked as *deprecated*, indicating that it will be
+removed/changed in the future.
+
+The major version is the only time all existing BC layers are removed. The last
+minor version before a new major version (i.e. 2.7 is the last minor version of
```

Comment:
do you know if this always will be the case? Will the second last minor version always contain the deprecation triggers and is the last minor version always released together with the new major version?